### ISSUE TITLE
Epoch length incorrectly calculated when using DDP

### ISSUE THREAD

**Issue by JAEarly** [mosaicml/composer#3444]
I'm using DDP with streaming, and it seems that the calculated epoch length is not correctly scaled by the world size.

With 16 GPUs, the console logger has outputs such as
```[epoch=11][batch=12910/834882]```
suggesting there are ~835K batches per epoch.

However, each epoch terminates correctly at ~52K batches, which is what I expect as it is the total number of batches / world size = ~835K / 16. This is also causing issues with the `LinearWithWarmupScheduler` set to warmup for 1 epoch - the LR is warming up 16x slower than it should be (i.e. it will warmup for 16 epochs rather than 1).

I think the epoch length calculations are not scaled by the world size, i.e. the trainer and scheduler believe the batch size is 64 rather than 64 * 16 = 1024. Note I am passing `batch_size=64` to the dataloader as this is the per-device batch size I want to be working at.

I am unsure if I have incorrectly configured something, or whether this is an underlying issue in composer and/or streaming. Please could you advise on whether this is expected to work correctly (i.e. there is something wrong on my end) or whether I should look into opening a PR to fix this. Thanks!

**Comment by mvpatel2000** [comment]
@JAEarly  would you mind adding streaming and composer versions? We've fixed a similar issue recently so want to confirm it's not that

**Comment by JAEarly** [comment]
> @JAEarly would you mind adding streaming and composer versions? We've fixed a similar issue recently so want to confirm it's not that

Sure, apologies for not providing those. 

composer: 0.23.2
streaming: 0.7.6

What version was the similar issue fixed in? I believe these are the correct versions; we may be using slightly older ones.

**Comment by mvpatel2000** [comment]
@JAEarly looks different.. can you print your dataloader length and compare to number of steps composer thinks? If you have a repro that would be helpful as well

**Comment by JAEarly** [comment]
> @JAEarly looks different.. can you print your dataloader length and compare to number of steps composer thinks? If you have a repro that would be helpful as well

Apologies for the delay. The dataloader length and number of steps per epoch is the same for both single GPU and distributed (2 x 8):

```
1 x 1:
  Train dataloader length: 834882
  Composer number of steps per epoch: 834882
 
2 x 8:
  Train dataloader length: 834882
  Composer number of steps per epoch: 834882
```

Note in both cases I'm passing batch size = 64 to the dataloader.

**Comment by JAEarly** [comment]
After further investigation, I've realised this was because we were using the "size" rather than "length" of the dataset.
I have confirmed this now fixes the issues and will close the issue.
Thank you for the input!
